component6/conversation_manager.py

```
@@ -76,495 +95,352 @@ async def process_conversation(
+        try:
+            # Use context assembler to create Gemini request
+            gemini_request = await self.context_assembler.assemble_context(
+                conversation_context=conversation_context,
+                current_message=user_message,
+                target_tokens=2000
+            )
+            
+            self.logger.debug(f"Assembled Gemini context with token budget: {gemini_request.token_budget}")
             return gemini_request
             
         except Exception as e:
```

Ruff (0.12.2)

Do not catch blind exception: `Exception`

(BLE001)